cmd/nanotube/listen.go

```diff
@@ -66,17 +72,86 @@ loop:
 	close(queue)
 }
 
-func readFromConnection(wg *sync.WaitGroup, conn net.Conn, queue chan string, stop <-chan struct{}, cfg *conf.Main, ms *metrics.Prom, lg *zap.Logger) {
+func listenUDP(cfg *conf.Main, queue chan string, stop <-chan struct{}, lg *zap.Logger, ms *metrics.Prom) {
+	// TODO (grzkv) What should we do if connection drops?
+	conn, err := net.ListenUDP("udp", &net.UDPAddr{
+		IP:   nil,
+		Port: int(cfg.ListeningPort),
+		Zone: ""})
+	if err != nil {
+		lg.Error("error while opening UDP port for listening", zap.Error(err))
+		return
```

Comment:
Crash with error message/return this error back.
Common use case would be when there is another instance running&listening on the same port. We don't want to have the new instance hanging around if it fails to Listen().
Better to crash and let systemd/other system restart.